velox/exec/Window.cpp

```diff
@@ -350,10 +388,50 @@ void Window::callApplyForPartitionRows(
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
   }
 
-  auto updateFrameBounds = [&](vector_size_t* rawFrameBounds,
+  auto readFrameValues = [&](const FrameChannelArg& frameArg) -> void {
+    frameArg.value->resize(numRows);
+    windowPartition_->extractColumn(
+        frameArg.index, partitionOffset_, numRows, 0, frameArg.value);
+  };
+
+  auto columnValuesCheck = [&](const int64_t* offsets) -> void {
+    if (!std::all_of(offsets, offsets + numRows, [&](vector_size_t i) {
+          return i > 0;
+        })) {
+      VELOX_FAIL("k in frame bounds must be at least 1");
+    }
+  };
+
+  auto updateKRowsFrameBounds = [&](vector_size_t* rawFrameBounds,
```

Comment:
List all input parameters before output parameters in function definitions. Please update all your function definitions accordingly.

Reply:
Apologies for missing the ordering here, it is addressed in remaining functions. In ```updateFrameBounds```, since the last parameter is optional, the penultimate parameter is the output.